daemon/create.go

```diff
@@ -16,6 +17,13 @@ func (daemon *Daemon) ContainerCreate(name string, config *runconfig.Config, hos
 		return "", warnings, err
 	}
 
+	// The check for a valid workdir path is made on the server rather than in the
+	// client. This is because we don't know the type of path (Linux or Windows)
+	// to validate on the client.
+	if config.WorkingDir != "" && !filepath.IsAbs(config.WorkingDir) {
```

Comment:
That's exactly where I've put the check! :) 

@ahmetalpbalkan The call stack for the error you have above is

(create.go) ContainerCreate  <-- my check here
(create.go) Create
(daemon.go) newContainer
(daemon.go) generateIdAndName
(daemon.go) reserveName  <-  Invalid container name error here

We don't want the check for working dir lower down in that above stack as there would be other side effects to unwind which result in an un-necessarily long code path. Hence, still is the right place for the check.